code/game/objects/items/devices/portable_recharger.dm

```diff
@@ -0,0 +1,337 @@
+/obj/item/portable_recharger
+	name = "backpack recharger"
+	icon = 'icons/obj/rechargers.dmi'
+	icon_state = "backcharger"
+	base_icon_state = "backcharger"
+	desc = "A portable backpack charging dock for energy based weaponry, PDAs, and other devices."
+	inhand_icon_state = "backcharger"
+	lefthand_file = 'icons/mob/inhands/equipment/security_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/equipment/security_righthand.dmi'
+	slot_flags = ITEM_SLOT_BACK
+	force = 5
+	throwforce = 6
+	w_class = WEIGHT_CLASS_BULKY
+	///Our power cell
+	var/obj/item/stock_parts/cell/cell
+	///What item is being charged currently?
+	var/obj/item/charging = null
+	///Did we put power into "charging" last process()?
+	var/using_power = FALSE
+	///Did we finish recharging the currently inserted item?
+	var/finished_recharging = FALSE
+	///The sound that it makes while charging
+	var/datum/looping_sound/recharger/soundloop
+	///Was it hit?
+	var/hit = FALSE
+	///What is the chance of it malfunctioning when shot or EMP'd?
+	var/malf_chance = 5
+
+	var/static/list/allowed_devices = typecacheof(list(
+		/obj/item/gun/energy,
+		/obj/item/melee/baton/security,
+		/obj/item/ammo_box/magazine/recharge,
+		/obj/item/modular_computer,
+	))
+
+/obj/item/portable_recharger/get_cell()
+	return cell
+
+/obj/item/portable_recharger/Initialize(mapload)
+	. = ..()
+	AddElement(/datum/element/drag_pickup)
+	soundloop = new(src)
+	update_appearance()
+
+/obj/item/portable_recharger/loaded/Initialize(mapload)
+	. = ..()
+	cell = new /obj/item/stock_parts/cell/high(src)
+
+/obj/item/portable_recharger/Destroy()
+	QDEL_NULL(soundloop)
+	return ..()
+
+/obj/item/portable_recharger/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot & (ITEM_SLOT_BACK|ITEM_SLOT_BELT))
```

Comment:
```suggestion
	if(slot & slot_flags)
```
This will automatically pass in back if it's the back one, or belt if it's the belt one